src/test/tst_attrsgenerator.cpp

```diff
@@ -31,61 +31,434 @@ class TestAttrsGenerator: public QObject
 private slots:
     void initTestCase() {}
     void cleanupTestCase() {}
-    void tst_parseStarCmd();
-    void tst_parseHashCmd();
+    void tst_parseStarCmd_min();
+    void tst_parseStarCmd_max();
+    void tst_parseStarCmd_rand();
+    void tst_parseHashCmd_min();
+    void tst_parseHashCmd_max();
+    void tst_parseHashCmd_rand();
+    void tst_parseHashCmd_setValue();
+    void tst_parseHashCmd_mixedFunc();
+    void _tst_attrs(SetOfAttributes res, Attributes attrs,  bool testValues);
```

Comment:
**pro-tip**: it's a good practice to always pass objects by a const reference unless we really need a copy of the object.

https://stackoverflow.com/a/2582813/3836404